CONTRIBUTING.md

```diff
@@ -1,75 +1,389 @@
-* Spellcheck - This checks the Markdown documents against a dictionary
-  and a local wordlist
-* Lint and Test Snippets - This tests the code snippets that are
-  imported into the documentation. Note that code included inline does
-  not get tested or run. 
+- Test the build
+  - Tests for build errors and does HTML5 validation. 
+- Lint and Test Snippets
+  - Tests the imported code snippets. Code included inline is **not** tested or run.
+- Spellcheck - **Does not work yet. Until finished, please check your docs in an external spellchecker before merging.**
+  - Checks Markdown files against a dictionary and a local wordlist. 
 
-## Spellcheck
+### Spellcheck
+
+**Does not work yet. Until finished, please check your docs in an external spellchecker before merging.**
 
 This is currently only available in Github Actions on Pull Requests.
-Any words that do not pass spell check will be shown in the logs of
+Any words that do fail spell check are shown in the logs of
 the Github Action. Please either fix the words or add them to the list
-in `Agoric/documentation/.wordlist.txt`. Please maintain the list's alphabetical order when entering new words for the convenience of future maintainers.
+in `Agoric/documentation/.wordlist.txt`. When entering new words, please keep the list 
+in alphabetical order for the convenience of future maintainers.
 
 ![](./contributing-assets/spellcheck-results.png)
 
-## Importing and testing snippets
+### Importing and testing code snippets
+
+**tyg todo: Note: There was a lot of this that I found possibly ambiguous. I've tried
+to specify a few more things to make it clearer, but you should probably review
+this section carefully make sure my interpretations are correct or not**
 
-To import code snippets into the documentation, you can write up the
-code in a separate file under `snippets/`, then include it by adding a
-line like `<<< @/snippets/test-intro-zoe.js#install` to your document.
-The section `test-intro-zoe.js` should be replaced with your
-particular filename. `#install` is an example of a specific code
-region in the file to import. For example, it might look like:
+Code snippets are not short inline code bits like `const x = 2 + 2;`. Or 
+code blocks denoted in Markdown by starting with a line consisting of three backquotes with 
+`js` appended and ending with a line consisting of three backquotes. 
 
+Rather, code snippets are actual development or test code from the Agoric-SDK repo,
+or code held to a similar standard of correctness. They should pass `lint` and run with no
+errors. This ensures our documents use real code that works
+with the current agoric-sdk version (whatever is on master) and is
+not outdated. **tyg todo: I'm not quite seeing this guaranteeing things,
+since we're not extracting the code directly from agoric-sdk. It'd seem
+if we don't check and manually update the snippets file after each upgrade, we risk
+falling out of sync with the agoric-sdk repo.**
```

Comment:
you might say it "provides assurance" to use instead of "guarantees", for the reasons you cite.